### PR TITLE
Integrate CircleCI

### PR DESCRIPTION
**Pull request by jochym**
- Redundant builders removed from travis config
- Additional builders configured on CircleCI: centos, fedora, opensuse (+debian and ubuntu) 
- Building dockers created for all above distros (could be used in local testing)
- Fixed few small problems in the process
- Dockerfiles could be used as examples to configure development environment
- auto-building docker images are hosted on dockerhub
- Stand-alone build/testing envs for all mentioned distros. You can run this to build on any distros from dockerhub repo (jochym/indi-*). For example to build on centos:
`docker run --rm -it jochym/indi-centos ./run-build.sh indilib master`
- If you have driver development branch with name `drv_name-of-the-driver` you can run the build of just the driver against current indi - this is limited to ubuntu images:
`docker run --rm -it jochym/indi-bionic ./run-build.sh jochym drv_gpsd`
- I think we should later limit the set of builds to just a few and build on all only for the releases

### CODE REVIEW

File: .circleci/debian_buster/Dockerfile [hunk 1]
```diff
@@ -11,3 +11,10 @@ RUN apt-get -qqy install \
         libboost-test-dev python-all-dev swig g++ libftdi1-dev \
         googletest google-mock clang
 
+WORKDIR /home
+
+RUN echo "#!/bin/bash" > /home/run-build.sh
```
[the remark this answers is not in the document]
Comment (jochym) [reply .circleci/debian_buster/Dockerfile]: You are right. I'll modify the dockers.

**Comment by knro** [comment]
I agree with you on the last point. We probably don't need to trigger complete build on 5 distros after each Git PR. And how do we selectively trigger builds (for all distros vs 2 for example)?

File: .circleci/debian_buster/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (TallFurryMan) [reply .circleci/debian_buster/Dockerfile]: Another way is to provide both script and permission configuration as a build context file, and ADD or COPY it in the image. Useful to keep code written once.

**Comment by TallFurryMan** [comment]
Given the difference of structure in libraries and include files in distributions, I'd tend to agree with @jochym as this would validate the build recipes. Also this would force devs to think ahead when adding a dependency, as docker files will need to be updated. But well, these are additional builds indeed, and probably not in the "supported list".

**Comment by TallFurryMan** [comment]
(my sentence is a bit weird around "agree" but I think you get the point :))

**Comment by jochym** [comment]
I have moved the filtering to the `build-all.sh` script so it is fairly elastic and it will be mostly independent of the CI platform we select in the future. The threshold is at 4 builds, the next is 8. This is the number of parallel jobs at CCI. The job at travis takes at least 12-14 min (often longer), at CCI 8min. I think we should start with all jobs to make sure everything is working as expected. And I think we should limit it after few days to speed up the build and not hog the resources at CCI. 
I also think that we should keep the dockers maintained (and possibly extended) as a good testing tool for the developers. Maybe I should add appropriate text to the docs? Also separate driver build chain needs to be made working on non-ubuntu machines (that is for later date).

**Comment by jochym** [comment]
I am making some adjustments. Dont merge yet  @knro  ;)

**Comment by jochym** [comment]
We need to wait for dockers to build before merging. Let me know @knro  if you decide to merge this and I will merge it myself - making sure the timing is right.

**Comment by knro** [comment]
Go ahead and merge it when it's good to go. Thanks!!

**Pull request merged**